zebra-network/src/peer_set/set.rs

```diff
@@ -496,7 +507,18 @@ where
                     // - always do the same checks on every ready peer, and
                     // - check for any errors that happened right after the handshake
                     trace!(?key, "got Change::Insert from Discover");
-                    self.remove(&key);
+
+                    // # Security
+                    //
+                    // drop the new peer if there are already `MAX_CONNS_PER_IP` peers with
+                    // the same IP address in the peer set
+                    // this is skipped for tests so we can mock peer connections
+                    #[cfg(not(test))]
```

Comment:
I just realised that disabling this check in tests will also disable it in integration and acceptance tests. So we'll still get the testnet failures we're seeing on other PRs in CI.

Reply:
Replaced with a configurable `max_conns_per_ip` in https://github.com/ZcashFoundation/zebra/pull/6980/commits/cbd9041affac75f684b41058379e17caa060743b.

Reply:
Thanks!